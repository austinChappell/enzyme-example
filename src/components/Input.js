// External Dependencies
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styled from 'styled-components';

// Internal Dependencies
import { COLORS } from '../constants/styles';

// Local Variables
const StyledInput = styled.input({
  border: `1px solid ${COLORS.primaryMain}`,
  borderRadius: 8,
  fontSize: 16,
  outline: 'none',
  padding: 8,
  transition: '200ms',
  '&:focus': {
    boxShadow: `
      1px 1px 4px ${COLORS.primaryLight},
      -1px -1px 4px ${COLORS.primaryLight},
      1px -1px 4px ${COLORS.primaryLight},
      -1px 1px 4px ${COLORS.primaryLight}
    `,
  }
});

// Component Definition
class Input extends Component {
  static propTypes = {
    ariaLabel: PropTypes.shape.isRequired,
    onChange: PropTypes.func.isRequired,
    placeholder: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string.isRequired,
  };

  static defaultProps = {
    placeholder: null,
    type: 'text',
  };

  render() {
    return (
      <StyledInput
        aria-label={this.props.ariaLabel}
        onChange={this.props.onChange}
        placeholder={this.props.placeholder}
        type={this.props.type}
        value={this.props.value}
      />
    );
  }
}

export default Input;
