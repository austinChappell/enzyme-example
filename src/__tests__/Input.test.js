import React from 'react';
import { mount, shallow } from 'enzyme';

import Input from '../components/Input';

describe('<Input />', () => {
  it('should match snapshot', () => {
    const mockChange = jest.fn();
    const wrapper = shallow(
      <Input
        ariaLabel="email"
        name="email"
        onChange={mockChange}
        value="Email"
      />
    );
    expect(wrapper).toMatchSnapshot();
  })

  it('should fire onChange when typing', () => {
    const mockChange = jest.fn();
    const wrapper = mount(
      <Input
        ariaLabel="email"
        name="email"
        onChange={mockChange}
        value="Email"
      />
    );
    wrapper.find('input').simulate('change');
    expect(mockChange).toHaveBeenCalled();
  })

  it('should fire onChange when triggered programmatically', () => {
    const mockChange = jest.fn();
    const wrapper = mount(
      <Input
        ariaLabel="email"
        name="email"
        onChange={mockChange}
        value="Email"
      />
    );
    wrapper.props().onChange();
    expect(mockChange).toHaveBeenCalled();
  })
})