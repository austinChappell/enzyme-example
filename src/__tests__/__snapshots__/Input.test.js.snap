// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    ariaLabel="email"
    name="email"
    onChange={[MockFunction]}
    placeholder={null}
    type="text"
    value="Email"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "aria-label": "email",
      "name": "email",
      "onChange": [MockFunction],
      "placeholder": null,
      "type": "text",
      "value": "Email",
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "border: 1px solid #5C61DE; border-radius: 8px; font-size: 16px; outline: none; padding: 8px; transition: 200ms; &:focus {
  box-shadow: 1px 1px 4px #7C96FF,
      -1px -1px 4px #7C96FF,
      1px -1px 4px #7C96FF,
      -1px 1px 4px #7C96FF;
}",
        ],
      },
      "displayName": "styled.input",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "input",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "aria-label": "email",
        "name": "email",
        "onChange": [MockFunction],
        "placeholder": null,
        "type": "text",
        "value": "Email",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "border: 1px solid #5C61DE; border-radius: 8px; font-size: 16px; outline: none; padding: 8px; transition: 200ms; &:focus {
  box-shadow: 1px 1px 4px #7C96FF,
      -1px -1px 4px #7C96FF,
      1px -1px 4px #7C96FF,
      -1px 1px 4px #7C96FF;
}",
          ],
        },
        "displayName": "styled.input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
